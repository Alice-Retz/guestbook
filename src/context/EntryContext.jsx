import { useContext, createContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

const useEntry = () => {
  const context = useContext(EntryContext);

  if (context === undefined) {
    throw new Error(
      'useEntry hook must be called within a EntryContext Provider'
    );
  }

  return context;
};

export { EntryProvider, useEntry };
