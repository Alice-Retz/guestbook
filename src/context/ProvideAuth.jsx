import { createContext, useState } from 'react';
import React from 'react';

export const authContext = createContext();

export function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (username, password) => {
    const loginWasSuccessful =
      username === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;

    if (loginWasSuccessful) setUser({ username });
    return loginWasSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}
