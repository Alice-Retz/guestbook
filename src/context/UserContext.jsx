import { useContext, createContext, useState, useMemo } from 'react';
import React from 'react';

export const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'useUser hook must be called within a UserContext Provider'
    );
  }

  return context;
};

export { UserProvider, useUser };
