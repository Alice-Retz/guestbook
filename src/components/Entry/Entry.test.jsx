import { render, screen } from '@testing-library/react';
import React from 'react';
import { EntryProvider } from '../../context/EntryContext';
import Entry from './Entry';

it('should render an entry', () => {
  const { container } = render(
    <EntryProvider>
      <Entry entry={{ name: 'Alice', message: 'Test' }} />
    </EntryProvider>
  );

  const entryName = screen.getByText('-Alice');

  expect(entryName).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
