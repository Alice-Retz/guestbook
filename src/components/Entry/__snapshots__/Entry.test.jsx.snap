// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an entry 1`] = `
<div>
  <div>
    <h2>
      -
      Alice
    </h2>
    <p>
      Test
    </p>
  </div>
</div>
`;
